src/decimal/mod.rs

```diff
@@ -285,17 +285,14 @@ macro_rules! dec_impl {
     (impl_trait_from_float; $($t:ty),*) => {$(
         impl<T, P> From<$t> for GenericDecimal<T, P>
         where
-            T: Clone + GenericInteger,
+            T: Copy + Clone + GenericInteger + FromPrimitive,
             P: Copy + GenericInteger + Into<usize> + From<u8>
         {
             fn from(value: $t) -> Self {
                 if value.is_nan () { return GenericDecimal::nan() };
                 if value.is_infinite () { return if value.is_sign_negative () { GenericDecimal::neg_infinity() } else { GenericDecimal::infinity() } };
 
-                /* TODO: without the String conversion (probably through .to_bits) */
-                let src = format! ("{:+}", value);
-
-                GenericDecimal::from_decimal_str(&src).unwrap_or_else(|_| GenericDecimal::nan())
+                GenericDecimal(GenericFraction::from(value), P::zero())
```

Comment:
The `P::zero()` is what I'm referring to in the PR comment.

Reply:
We might use `calc_precision` method to calculate it, or figure out how to extract it from the source float?